frontend/app/services/work-package-service.js

```diff
@@ -221,9 +261,11 @@ module.exports = function($http,
     },
 
     updateWorkPackage: function(workPackage, notify) {
+      console.log(workPackage.links.updateImmediately);
       var options = { ajax: {
-        method: 'PATCH',
-        url: URI(workPackage.links.updateImmediately.href).addSearch('notify', notify).toString(),
+        method: workPackage.links.updateImmediately.props.method,
+        url: URI(workPackage.links.updateImmediately.props.href)
+            .addSearch('notify', notify).toString(),
```

Comment:
I think I see the reason why the WP is initialized with bogus `update` links here.

Hint: The work package form you are working with also always has a link called `commit`. Depending on whether it is a create form or an update form, the links are different. In either case they should also indicate the correct method.
This knowledge might come in handy when refactoring this code.